bundles/org.palladiosimulator.dataflow.confidentiality.analysis/src/org/palladiosimulator/dataflow/confidentiality/analysis/sequence/entity/AbstractActionSequenceElement.java

```diff
@@ -8,20 +8,27 @@
 public abstract class AbstractActionSequenceElement<T extends EObject> {
 
     private final Optional<List<DataFlowVariable>> dataFlowVariables;
+    private final Optional<List<CharacteristicValue>> nodeVariables;
 
     public AbstractActionSequenceElement() {
         this.dataFlowVariables = Optional.empty();
+        this.nodeVariables = Optional.empty();
     }
 
-    public AbstractActionSequenceElement(List<DataFlowVariable> variables) {
-        this.dataFlowVariables = Optional.of(List.copyOf(variables));
+    public AbstractActionSequenceElement(List<DataFlowVariable> dataFlowVariables, List<CharacteristicValue> nodeVariables) {
+        this.dataFlowVariables = Optional.of(List.copyOf(dataFlowVariables));
+        this.nodeVariables = Optional.of(List.copyOf(nodeVariables));
     }
 
     public abstract AbstractActionSequenceElement<T> evaluateDataFlow(List<DataFlowVariable> variables);
-
+    
     public List<DataFlowVariable> getAllDataFlowVariables() {
         return this.dataFlowVariables.orElseThrow(IllegalStateException::new);
     }
+    
+    public List<CharacteristicValue> getAllNodeVariables() {
```

Comment:
I would propose naming this `getAllNodeCharacteristics` as a node does not has variables. The node has characteristics and manages data flow variables that also have characteristics (that we then compare in the constraints).

Reply:
I agree. As these CharacteristicValues do not change, it would be more precise to call them Node Characteristics. As such, i have renamed occurences in the recent commit